xendit-java-lib/src/main/java/com/xendit/model/DisbursementClient.java

```diff
@@ -130,25 +130,89 @@ public AvailableBank[] getAvailableBanks(Map<String, String> headers) throws Xen
         RequestResource.Method.GET, url, headers, null, opt.getApiKey(), AvailableBank[].class);
   }
 
-  public Disbursement[] getByExternalId(String externalId) throws XenditException {
+  public DisbursementIDR[] getByExternalId(String externalId) throws XenditException {
     return getByExternalId(new HashMap<>(), externalId);
   }
 
-  public Disbursement[] getByExternalId(Map<String, String> headers, String externalId)
+  public DisbursementIDR[] getByExternalId(Map<String, String> headers, String externalId)
       throws XenditException {
     String url =
         String.format("%s%s%s", opt.getXenditURL(), "/disbursements?external_id=", externalId);
     return this.requestClient.request(
-        RequestResource.Method.GET, url, headers, null, opt.getApiKey(), Disbursement[].class);
+        RequestResource.Method.GET, url, headers, null, opt.getApiKey(), DisbursementIDR[].class);
   }
 
-  public Disbursement getById(String id) throws XenditException {
+  public DisbursementPH[] getByReferenceId(Map<String, String> headers, String referenceId)
+      throws XenditException {
+    String url =
+        String.format("%s%s%s", opt.getXenditURL(), "/disbursements?reference_id=", referenceId);
+    return this.requestClient.request(
+        RequestResource.Method.GET, url, headers, null, opt.getApiKey(), DisbursementPH[].class);
+  }
+
+  public DisbursementIDR getById(String id) throws XenditException {
     return getById(new HashMap<>(), id);
   }
 
-  public Disbursement getById(Map<String, String> headers, String id) throws XenditException {
+  public DisbursementIDR getById(Map<String, String> headers, String id) throws XenditException {
+    String url = String.format("%s%s%s", Xendit.Opt.getXenditURL(), "/disbursements/", id);
+    return this.requestClient.request(
+        RequestResource.Method.GET, url, headers, null, opt.getApiKey(), DisbursementIDR.class);
+  }
+
+  public DisbursementPH getPHById(Map<String, String> headers, String id) throws XenditException {
     String url = String.format("%s%s%s", Xendit.Opt.getXenditURL(), "/disbursements/", id);
     return this.requestClient.request(
-        RequestResource.Method.GET, url, headers, null, opt.getApiKey(), Disbursement.class);
+        RequestResource.Method.GET, url, headers, null, opt.getApiKey(), DisbursementPH.class);
+  }
+
+  public DisbursementPH createPHRequest(Map<String, String> headers, Map<String, Object> params)
```

Comment:
@hglohika you need to do the same thing for PH, which is to provide overloaded methods of `create()`.  but since we got two different ones: one for IDR and one for PH, we need to rename.


`createPHDisbursement`
`createIDRDisbursement` (rename the current IDR one)
are public methods, and are overloaded. you can see the current `create` one is overloaded for **each optional field**

both `createPHDisbursement` and `createIDRDisbursement` should call `createRequest`. Inside `createRequest`, do a check on the keys provided in the params.

if it contains `currency` = PH, use `DisbursementPH` model as class, which should be the returned instance type of the request. `DisbursementPH.class`. Else use the `DisbursementIDR` model as the returned instance type of the request.

Reply:
a674639a02e336bbe43b3beb4af3e8ab1db807c1

Reply:
as discussed via dm:
```
if it contains currency = PH, use DisbursementPH model as class, which should be the returned instance type of the request. DisbursementPH.class. Else use the DisbursementIDR model as the returned instance type of the request.
```
is not possible because java has static typing, and requires a fixed type to return. Updating this to generics is a breaking change, so leaving it like this for now